### PR TITLE
RUST-1810 Clean up concern helpers

### PR DESCRIPTION
**Pull request by abr-egn**

### CODE REVIEW

File: src/concern.rs [hunk 1]
```diff
@@ -104,8 +81,8 @@ impl ReadConcern {
     /// Creates a read concern with a custom read concern level. This is present to provide forwards
     /// compatibility with any future read concerns which may be added to new versions of
     /// MongoDB.
-    pub fn custom(level: String) -> Self {
-        ReadConcernLevel::from_str(level.as_str()).into()
+    pub fn custom(level: &str) -> Self {
```
Comment (isabelatkinson) [inline src/concern.rs:84]:
```suggestion
    pub fn custom(level: impl AsRef<str>) -> Self {
```

File: src/concern.rs [hunk 2]
```diff
@@ -322,6 +295,21 @@ impl From<String> for Acknowledgment {
...
+        Acknowledgment::Nodes(v).into()
+    }
+
+    /// A `WriteConcern` requesting [`Acknowledgment::Majority`].
+    pub fn majority() -> Self {
+        Acknowledgment::Majority.into()
+    }
+
+    /// A `WriteConcern` with a custom acknowledgment.
+    pub fn custom(s: &str) -> Self {
```
Comment (isabelatkinson) [inline src/concern.rs:309]:
```suggestion
    pub fn custom(s: impl AsRef<str>) -> Self {
```

File: src/concern.rs (hunk 1 above)
Comment (abr-egn) [reply src/concern.rs]: Updated.

File: src/concern.rs (hunk 2 above)
Comment (abr-egn) [reply src/concern.rs]: Updated.

**Review by isabelatkinson — APPROVED** [review]

**Pull request merged**